plugins/woocommerce/includes/admin/importers/class-wc-product-csv-importer-controller.php

```diff
@@ -113,37 +114,18 @@ public static function is_file_valid_csv( $file, $check_path = true ) {
 	 * @throws \Exception When file validation fails.
 	 */
 	protected static function check_file_path( string $path ): void {
-		$is_valid_file = false;
+		$wp_filesystem = FilesystemUtil::get_wp_filesystem();
 
-		if ( ! empty( $path ) ) {
-			$path          = realpath( $path );
```

Comment:
I'm keen to pause and be extra sure this is safe before we continue without a suitable replacement for `realpath`, because of the risk of directory traversals. Imagine that:

- WordPress is installed in the public web root at `/srv/public/`.
- Existing readable file `/srv/secret.csv` is outside of the web root.
- Request var `file` (which `$path` is based on) is set to something like `/srv/public/wp-content/uploads/../../../secret.csv`.

If we don't (or cannot) use `realpath()` to resolve the path, we won't detect the directory traversal. New private method `file_is_in_directory` only performs a basic string comparison, which also won't detect this.

I've sometimes advocated for the use of `Automattic\WooCommerce\Internal\Utilities\URL` for situations like this (despite the name, it can indeed work with filepaths). Whether we use that specific utility here or not, though, I think we may need a similar capability in the sense of being able to resolve directory traversals without using `realpath`.

(Alternatively, perhaps WP provides a method for this in its file system abstraction, though I couldn't immediately identify anything like that.)